pajbot/bot.py

```diff
@@ -130,16 +120,44 @@ def __init__(self, config: cfg.Config, args: argparse.Namespace) -> None:
         self.app_token_manager = AppAccessTokenManager(self.twitch_id_api, RedisManager.get())
         self.twitch_helix_api: TwitchHelixAPI = TwitchHelixAPI(RedisManager.get(), self.app_token_manager)
 
-        self.bot_user_id = self.twitch_helix_api.get_user_id(self.nickname)
+        # bot
+        if "bot_id" in config["main"]:
+            self.bot_user_id = config["main"]["bot_id"]
+            self.nickname = self.twitch_helix_api.get_login(self.bot_user_id)
+        elif "nickname" in config["main"]:
+            self.nickname = config["main"].get("nickname", "pajbot")
+            self.bot_user_id = self.twitch_helix_api.get_user_id(self.nickname)
```

Comment:
if bot_id and nickname is missing, we silently fail and will error out in a weird way later

Reply:
You should use `require_user_id` which already exists in the helix api wrapper: https://github.com/pajbot/pajbot/blob/e28d12cc430ff7567208a439d1c88a5204ef6378/pajbot/apiwrappers/twitch/helix.py#L223-L227